.github/workflows/build.yml

```diff
@@ -22,8 +22,14 @@ jobs:
       - name: Install Rust
         uses: dtolnay/rust-toolchain@stable
         with:
-          toolchain: nightly # nightly is required for fmt
+          toolchain: nightly
           components: rustfmt, clippy
+          targets: aarch64-linux-android armv7-linux-androideabi i686-linux-android x86_64-linux-android
+
+      - name: Set up Cargo Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          save-if: false
```

Comment:
@Pushkarm029 I think that's a great idea. Just curious, where the cache will be really stored in this case?

@oluiscabral here is our weekly job, could you enable cache storage in it?
https://github.com/ARK-Builders/ark-core/actions/runs/11991530925

Reply:
Also, do I understand correctly that we should not store cache after running branch builds because it can be incompatible with other branches? Am I right that we still should save cache when a new commit is merged into `main`?

Reply:
@kirillt I'm not 100% sure what you mean by `because it can be incompatible with other branches`, but I see no problem storing branch build cache, considering that according to their README (https://github.com/Swatinem/rust-cache) unrelated branches don't affect each other:
![image](https://github.com/user-attachments/assets/3cf2cd72-b52c-45f7-b7eb-a8d99d0d00f4)

Reply:
> where the cache will be really stored in this case?

The cache is stored on GitHub servers. You can view the cached entries by navigating to the Cache section on the left-hand side of the Actions tab in this repository.

> because it can be incompatible with other branches?

No, that is not possible. You can read more about this [here](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache).

I recommended storing the cache on the `main` branch because:
- It prevents potential cache poisoning from experimental dependencies in (same) PR.
- Multiple PRs can benefit from a shared cache built from the main branch.

If each PR created its own cache, it would result in duplicate caches for common dependencies, leading to faster exhaustion of GitHub Actions storage.

>  Am I right that we still should save cache when a new commit is merged into main?


Yes, saving the cache when a new commit is merged into main is more efficient than doing so on a weekly basis.